cdserver/config/application.rb

```diff
@@ -12,6 +12,8 @@
 # require "sprockets/railtie"
 require "rails/test_unit/railtie"
 
+require_relative "../lib/middleware/middleware_healthcheck"
```

Comment:
According to my reading, lib/middleware should be auto-loaded.  You shouldn't need to require the file explicitly.  See here: http://ieftimov.com/writing-rails-middleware
It looks like you have to use strings due to the order of loading not guaranteeing the constants are loaded.
Give it a try see if you can make it work